pedpy/methods/acceleration_calculator.py

```diff
@@ -0,0 +1,346 @@
+
+        Args:
+            message: Error message
+        """
+        self.message = message
+
+
+def compute_individual_acceleration(
+    *,
+    traj_data: TrajectoryData,
+    frame_step: int,
+    movement_direction: Optional[npt.NDArray[np.float64]] = None,
+    compute_acceleration_components: bool = False,
+    acceleration_calculation: AccelerationCalculation = AccelerationCalculation.BORDER_EXCLUDE,
+) -> pandas.DataFrame:
+    r"""Compute the individual acceleration for each pedestrian.
+
+    For computing the individuals' acceleration at a specific frame :math:`a_i(t_k)`,
+    a specific frame step (:math:`n`) is needed.
+    Together with the
+    :attr:`~trajectory_data.TrajectoryData.frame_rate` of
+    the trajectory data :math:`fps` the time frame :math:`\Delta t` for
+    computing the speed becomes:
+
+    .. math::
+
+        \Delta t = 2 n / fps
+
+    This time step describes how many frames before and after the current
+    position :math:`X(t_k)` are used to compute the movement.
+    These positions are called :math:`X(t_{k+n})`, :math:`X(t_{k-n})`
+    respectively.
+
+    In order to compute the acceleration at time 't_k', we first calculate the
+    displacements :math:`\bar{X}` around 't_{k+n}' and 't_{k-n}':
+
+    .. math::
+
+        \bar{X}(t_{k+n}) = X(t_{k+2n}) - X(t_{k})
+
+    .. math::
+
+        \bar{X}(t_{k-n}) = X(t_{k}) - X(t_{k-2n})
+
+    The acceleration is then calculated from the difference of the displacements
+
+    .. math::
+
+        \Delta\bar{X}(t_k) = \bar{X}(t_{k+n}) - \bar{X}(t_{k-n})
+
+    divided by the square of the time interval '\Delta t':
+
+    .. math::
+
+        a_i(t_k) = \Delta\bar{X}(t_k) / \Delta t**2
+
+
+    When getting closer to the start, or end of the trajectory data, it is not
+    possible to use the full range of the frame interval for computing the
+    acceleration. For these cases *PedPy* offers a method to compute
+    the acceleration:
+
+    **Exclude border:**
+
+    When not enough frames available to compute the speed at the borders, for
+    these parts no acceleration can be computed and they are ignored. Use
+    :code:`acceleration_calculation=AccelerationCalculation.BORDER_EXCLUDE`.
+
+
+    **With movement direction:**
+
+    It is also possible to compute the individual acceleration in a specific direction
+    :math:`d`, for this the movement :math:`\Delta\bar{X}` is projected onto the
+    desired movement direction. :math:`\Delta\bar{X}` and :math:`\Delta t` are
+    computed as described above. Hence, the acceleration then becomes:
+
+    .. math::
+
+        a_i(t) = {{|\boldsymbol{proj}_d\; \Delta\bar{X}|} \over {\Delta t**2}}
+
+
+    If :code:`compute_acceleration_components` is `True` also :math:`\Delta\bar{X}` is returned.
+
+    Args:
+        traj_data (TrajectoryData): trajectory data
+        frame_step (int): gives the size of time interval for calculating the
+            velocity.
```

Comment:
I think "velocity" here should be replace by "acceleration".

Reply:
done